packages/flutter_tools/lib/src/project.dart

```diff
@@ -251,6 +255,40 @@ class FlutterProject {
   }
 }
 
+/// Base class for projects per platform.
+abstract class FlutterProjectPlatform {
+
+  /// Plugin's platform config key, i.e., "macos", "ios".
+  String get pluginConfigKey;
+
+  /// Whether a the platform exists in the project.
+  bool existsSync();
+
+  /// Creates a list with the project plugins supported for the current platform.
+  List<Map<String, dynamic>> pluginsList(List<Plugin>plugins) {
```

Comment:
`filterPlugins`? The current name suggests to me that it's returning a list of plugins that are a property of the project, rather than being driven by the input list. (It feels a bit odd for this to be part of the project, rather than a stand-alone utility or something on a plugins class, but I don't feel super strongly about that.)

Reply:
I agree it feels better as a helper function. Done